core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -1345,7 +1345,7 @@ private QueryModel parseDmlUpdate(GenericLexer lexer, SqlParserCallback sqlParse
         fromModel.setModelPosition(modelPosition);
         updateQueryModel.setIsUpdate(true);
         fromModel.setIsUpdate(true);
-        tok = tok(lexer, "UPDATE, WITH or table name expected");
+        tok = parseTableNameSansPublicSchema(lexer, "UPDATE, WITH or table name expected");
```

Comment:
this is very aggressive approach. Virtually everywhere we use to just get a token, now there is mostly redundant and quite unreadable table name parsing

Reply:
ok, there are ways to tone it down a bit. let me have a look if that results in readability improvements.

Reply:
I changed it so the stripping is only applied when we do actually expect a table name. wdyt?